### DISCUSSION TITLE
Possible to use keypad for arm/disarm?

### DISCUSSION THREAD

**Discussion by jaip-x** [tsightler/ring-mqtt#833] [General Discussion]
hi - integration is working well so far! Is there anyway to use a keypad instead of the "arm" or "disarm" buttons? Does Alarm Control Panel work with this Ring MQTT integration?

**Comment by tsightler** [comment]
I'm not really sure what you mean, Alarm Control Panel in HA is just Arm/Disarm button, with optional number entry pad if you enable disarm code.  This is already supported if you define a disarm code in the config.

If the question is, can the Alarm Control Panel automatically use the existing codes programmed in Ring for the keypad, then the answer is no, you have to setup separate codes in HA.

**Reply by jaip-x** [reply]
Ah, thanks for clarifying! I'm a noob, mind pointing me to the documentation of how to define the disarm code in the config? 

I'd like to use a number keypad not buttons.

**Reply by tsightler** [reply]
Not to sound too much like a, well, you know, but did you even look at the config?  Literally there's a field labeled "disarm_code", just put the code there and restart the addon, that's it.

**Comment by jaip-x** [comment]
No, I'm new to this.  Thanks for the direction!

--
*Jai Patel*
Founder, JUV <https://www.getjuv.com>
(m) 408.858.2026 <https://+1408.858.2026>
